platform/ios/app/MBXStateManager.h

```diff
@@ -0,0 +1,20 @@
+#import <Foundation/Foundation.h>
+#import "MBXState.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MBXStateManager : NSObject
+
+@property (nonatomic, nullable) MBXState* currentState;
+
++ (instancetype) sharedManager;
+
+- (void)saveState:(MBXState*)state forKey:(NSObject*)key;
```

Comment:
If we move the load/save logic here, then `-saveState:forKey:` could just become `saveState:` (or maybe even just `saveState` if the manager maintains a current object?)